client/my-sites/plugins/search-box-header/index.jsx

```diff
@@ -1,12 +1,16 @@
 import Search from '@automattic/search';
 import { useTranslate } from 'i18n-calypso';
-import { Fragment, useCallback, useEffect } from 'react';
+import { Fragment, useEffect } from 'react';
 import { useDispatch } from 'react-redux';
 import { setQueryArgs } from 'calypso/lib/query-args';
 import { recordGoogleEvent, recordTracksEvent } from 'calypso/state/analytics/actions';
 import './style.scss';
 
-const SearchBox = ( { isMobile, searchTerm, doSearch, searchBoxRef, isSearching } ) => {
+function pageToSearch( s ) {
+	setQueryArgs( '' !== s ? { s } : {} );
+}
```

Comment:
@candy02058912 this is the new `doSearch`. Youll be able to drop the window scroll call in here safely once its merged.